src/main/java/com/learning/browsers/BrowserDriverGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.learning.browsers;
+
+import com.learning.configuration.PropertiesReader;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+
+public class BrowserDriverGenerator implements BrowserDriver {
+
+  private static final String BROWSER_PROPERTIES_NAME = "browser";
+  private static final String PATH_TO_EDGE_DRIVER = "pathToEdgeDriver";
+  private final PropertiesReader propertiesReader;
+
+  public BrowserDriverGenerator() {
+    propertiesReader = new PropertiesReader();
+  }
+
+  public WebDriver getDriver() {
```

Comment:
This method will create new driver object each time it's called.
Instead create WebDriver field, and then:
```
if (driver == null) {
    //same if you already have, but instead return, assign driver to field
}
return driver;
```

This will allow you to create and manage only one instance of driver.

Reply:
OK I get it. We make sure here to have Singleton. Let me implement it. 
This will be a good exercise.

Reply:
If method `getDriver` gets driver from the filed is it enough?